### PR TITLE
Added info and examples for new functionalities 

### PR DESCRIPTION
**Pull request by manavellamnimble**

### CODE REVIEW

File: docs/source/analyze/node-resources.md [hunk 1]
```diff
@@ -124,3 +125,25 @@ spec:
...
+        filters:
+          cpuCapacity: "2"
+          selector: 
+            matchLabel: 
+               kubernetes.io/hostname: docker-desktop
+               kubernetes.io/role: database-primary-replica
+        outcomes:
+        - fail:
+            when: "count() < 3"
+            message: This application requires at least 3 nodes.
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:141]:
```suggestion
            message: This application requires at least 3 nodes, with 5 recommended.
```

File: docs/source/analyze/node-resources.md [hunk 2]
```diff
@@ -124,3 +125,25 @@ spec:
           - pass:
               message: This cluster has a node with enough memory.
 ```
+
+```yaml
+    - nodeResources:
+        checkName: Must have at least 3 nodes with 2 cores and labels kubernetes.io/hostname=docker-desktop and kubernetes.io/role=database-primary-replica
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:131]: I wouldn't suggest putting the labels in the checkName; these should be human-readable. Maybe:

```suggestion
        checkName: Must have at least 3 nodes with 2 cores, with Docker Desktop and a database
```

Comment (markpundsack) [reply docs/source/analyze/node-resources.md:131]: I don't understand how this check actually works. Where's the check for the labels? Does it filter the nodes by ones with those labels such that count() < 3 means there must be 3 nodes with docker-desktop on them?

File: docs/source/collect/copy.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Copy Files
+title: Copy Files and Folders
 description: A description of the copy collector to copy files out of pods
 ---
 
-The `copy` collector can be used to copy files from pods and include the contents in the collected data.
-This collector can be included multiple times to copy different files from different pods.
+The `copy` collector can be used to copy files or an etire folder from pods and include the contents in the collected data.
```
Comment (markpundsack) [inline docs/source/collect/copy.md:6]:
```suggestion
The `copy` collector can be used to copy files or an entire folder from pods and include the contents in the collected data.
```

File: docs/source/collect/copy.md [hunk 4]
```diff
@@ -36,11 +36,19 @@ metadata:
   name: sample
 spec:
   collectors:
+  #Copies resolv.conf file
     - copy:
-        selector: app=api
+        selector: 
+          - app=api
```
Comment (markpundsack) [inline docs/source/collect/copy.md:42]: Is the switch to an array necessary here?

Comment (manavellamnimble) [reply docs/source/collect/copy.md:42]: Actually this was not a change I have made. When I used the copy collector it worked using a list under the selector. Looking it closely now, the field is defined as a list of strings in the Copy struct:
```golang
type Copy struct {
	CollectorMeta `json:",inline" yaml:",inline"`
	Name          string   `json:"name,omitempty" yaml:"name,omitempty"`
	Selector      []string `json:"selector" yaml:"selector"`
	Namespace     string   `json:"namespace" yaml:"namespace"`
	ContainerPath string   `json:"containerPath" yaml:"containerPath"`
	ContainerName string   `json:"containerName,omitempty" yaml:"containerName,omitempty"`
}
```

Comment (manavellamnimble) [reply docs/source/collect/copy.md:42]: Actually I made a PR suggesting the change when that happened to me. I've closed it now.

File: docs/source/analyze/node-resources.md (hunk 2 above)
Comment (manavellamnimble) [reply docs/source/analyze/node-resources.md]: The main Idea is to be sure that the nodes being analyzed have certain labels. Maybe the example I used was miss-leading, check the new one in the new commit. In this case, for a node to match the filter this conditions must be met:
- Have 2 cores
- Have a label "kubernetes.io/hostname"="docker-desktop"
- Have a label "kubernetes.io/role"="database-primary-replica"

File: docs/source/collect/copy.md (hunk 4 above)
Comment (markpundsack) [reply docs/source/collect/copy.md:42]: So, just to be clear, are you saying that we don't actually support a direct string here at all, and we've always required it to be an array, and the example was just wrong, so now it's fixed?

File: docs/source/analyze/node-resources.md (hunk 2 above)
Comment (markpundsack) [reply docs/source/analyze/node-resources.md]: OK, so it sounds like all filters get applied, so it means that we need 3 nodes that have those labels and minimum cores, etc. That makes sense, but frankly is a bit confusing. If you're going to go that route, you need to change the actual messages. Currently, someone could have 3 nodes, but you'll tell them they don't, just because they don't have 3 nodes **with** those labels. This might be better as a different example, or even separate examples.

File: docs/source/collect/copy.md (hunk 4 above)
Comment (manavellamnimble) [reply docs/source/collect/copy.md:42]: Yes, exactly! With a string you would get:
Error: failed to parse collector: failed to parse document: v1beta1.SupportBundle.Spec: v1beta1.SupportBundleSpec.Collectors: []*v1beta1.Collect: v1beta1.Collect.Copy: v1beta1.Copy.Selector: []string: ...

Comment (markpundsack) [reply docs/source/collect/copy.md:42]: Perfect, thankjs.

File: docs/source/analyze/node-resources.md (hunk 2 above)
Comment (manavellamnimble) [reply docs/source/analyze/node-resources.md]: I've written a new, simpler example, only using the selector matchLabel (see last commit). Let me know what you think

File: docs/source/analyze/node-resources.md [hunk 5]
```diff
@@ -124,3 +125,24 @@ spec:
           - pass:
               message: This cluster has a node with enough memory.
 ```
+
+```yaml
+    - nodeResources:
+        checkName: Must have at least 3 nodes with labels beta.kubernetes.io/os=linux and kubernetes.io/rol=database-primary-replica
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:131]: Can you describe what these labels mean in plain English?
Also, is there a typo in `rol=`?

**Comment by manavellamnimble** [comment]
@markpundsack Hi Mark! I added a little description to the example, and used a use-case presented in the issue related to this enhancement. Let me know what you think!

**Comment by markpundsack** [comment]
@manavellamnimble Thanks. The `matchlabel` example seems more technically accurate (although still with a typo on `rol`), but my concern is that it's a poor user experience because they aren't describing what the labels mean. I wouldn't actually recommend a customer use a message like this. They need to be translated into plain English.

**Comment by manavellamnimble** [comment]
@markpundsack I modify the example using plain english to describe to the user what the label means. I send it now because I saw the team is working on the docs. If there is something else to modify let me know!

File: docs/source/analyze/node-resources.md [hunk 6]
```diff
@@ -124,3 +125,23 @@ spec:
           - pass:
               message: This cluster has a node with enough memory.
 ```
+### Filter by labels
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:128]:
```suggestion
### Filter by labels

> Filtering by labels was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.

```

File: docs/source/collect/copy.md [hunk 7]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Copy Files
+title: Copy Files and Folders
 description: A description of the copy collector to copy files out of pods
 ---
 
-The `copy` collector can be used to copy files from pods and include the contents in the collected data.
-This collector can be included multiple times to copy different files from different pods.
+The `copy` collector can be used to copy files or an entire folder from pods and include the contents in the collected data.
```
Comment (markpundsack) [inline docs/source/collect/copy.md:6]:
```suggestion
> The ability to copy folders was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.

The `copy` collector can be used to copy files or an entire folder from pods and include the contents in the collected data.
```

File: docs/source/analyze/node-resources.md [hunk 8]
```diff
@@ -124,3 +125,26 @@ spec:
...
+
+> Filtering by labels was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.
+
+Labels are intended to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system. Labels can be used to organize and to select subsets of objects.
+Troubleshoot allows users to analyze nodes that match one or more labels. For example, to require a certain number of nodes with certain labels as a preflight check, e.g. to check if a node has the label ```kubernetes.io/role=database-primary-replica```, indicating that it is running the Mongo Operator. Further filters may be applied to the nodes matching the labels. 
+
+```yaml
+    - nodeResources:
+        checkName: Must have 1 node with 16 GB (available) memory and 5 cores (on a single node) running Mongo Operator (label kubernetes.io/role=database-primary-replica expected in node).
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:137]:
```suggestion
        checkName: Must have at least 1 node running Mongo Operator (label kubernetes.io/role=database-primary-replica) with 16 GB (available) memory and 5 cores (on a single node).
```

File: docs/source/analyze/node-resources.md [hunk 9]
```diff
@@ -124,3 +125,26 @@ spec:
           - pass:
               message: This cluster has a node with enough memory.
 ```
+### Filter by labels
+
+> Filtering by labels was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.
+
+Labels are intended to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system. Labels can be used to organize and to select subsets of objects.
+Troubleshoot allows users to analyze nodes that match one or more labels. For example, to require a certain number of nodes with certain labels as a preflight check, e.g. to check if a node has the label ```kubernetes.io/role=database-primary-replica```, indicating that it is running the Mongo Operator. Further filters may be applied to the nodes matching the labels.
```
Comment (markpundsack) [inline docs/source/analyze/node-resources.md:133]:
```suggestion
Troubleshoot allows users to analyze nodes that match one or more labels. For example, to require a certain number of nodes with certain labels as a preflight check. Multiple filters may be specified and all are required to match for the node to match.
```

File: docs/source/analyze/node-resources.md (hunk 8 above)
Comment (markpundsack) [reply docs/source/analyze/node-resources.md:137]: Actually, the checkName should probably be simpler. Just Must have Mongo running. Sor something like that. The details could be in the fail message.

Comment (markpundsack) [reply docs/source/analyze/node-resources.md:137]:
```suggestion
        checkName: Must have Mongo running
```
